near-sdk/src/collections/tree_map.rs

```diff
@@ -580,6 +580,36 @@ where
     }
 }
 
+#[cfg(feature = "expensive-debug")]
+impl<K, V> std::fmt::Debug for TreeMap<K, V>
+where
+    K: std::fmt::Debug + Ord + Clone + BorshSerialize + BorshDeserialize,
+    V: std::fmt::Debug + BorshSerialize + BorshDeserialize,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TreeMap")
+            .field("root", &self.root)
+            .field("tree", &self.tree.iter().collect::<Vec<Node<K>>>())
+            .finish()
+    }
+}
```

Comment:
```suggestion
impl<K, V> std::fmt::Debug for TreeMap<K, V>
where
    K: std::fmt::Debug + Ord + Clone + BorshSerialize + BorshDeserialize,
    V: std::fmt::Debug + BorshSerialize + BorshDeserialize,
{
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.debug_struct("TreeMap")
            .field("root", &self.root)
            .field("tree", &self.tree)
            .finish()
    }
}
```

You need the debug impl for when the feature is not `expensive-debug`. There is no need to manually collect the nodes, the `Vector` debug implementation will do this in `expensive-debug` and do the lighter impl when it is not.

Reply:
Hmm, the problem with this is that the tree elements won't be displayed in a sorted order (which I personally would expect from the Debug impl for TreeMap). Alternatively, I could do something like this:
```suggestion
#[cfg(feature = "expensive-debug")]
impl<K, V> std::fmt::Debug for TreeMap<K, V>
where
    K: std::fmt::Debug + Ord + Clone + BorshSerialize + BorshDeserialize,
    V: std::fmt::Debug + BorshSerialize + BorshDeserialize,
{
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.debug_struct("TreeMap")
            .field("root", &self.root)
            .field("elements", &self.to_vec())
            .finish()
    }
}
```

And derive a default Debug impl when expensive-debug is not enabled. What do you think? The downside is that we are losing information about tree's structure (as the actual nodes won't get printed anymore).

Reply:
I mean, if you wanted to construct an in memory data structure that actually represents the data, you would collect a map (`BTreeMap<K, V>`) from the `TreeMap` iterator. You could optimize this by either iterating all keys unordered since BTreeMap enforces ordering or using some collection which keeps insertion order. This is all for expensive debug so I don't think it matters much the performance.

There aren't any strict rules around how the data is formatted, and I assume expensive debug was just added for very specific use cases, so this doesn't matter that much.

Reply:
Ok, I see your point and I see now that it probably makes more sense to format everything in a human-readable way that is close to the collections' internal structure rather than trying to come up with something more general.